android/src/com/frostwire/android/gui/activities/SettingsActivity.java

```diff
@@ -657,12 +658,32 @@ protected void onPostExecute(Void result) {
     @Override
     public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preference preference) {
 
-        boolean r = super.onPreferenceTreeClick(preferenceScreen, preference);
+        // If the user has clicked on a preference screen, set up the screen
         if (preference instanceof PreferenceScreen) {
-            initializePreferenceScreen((PreferenceScreen) preference);
-            currentPreferenceKey = preference.getKey();
+            setUpNestedScreen((PreferenceScreen) preference);
         }
-        return r;
+
+        return false;
+    }
+
+    public void setUpNestedScreen(PreferenceScreen preferenceScreen) {
+        final Dialog dialog = preferenceScreen.getDialog();
+
+        Toolbar nestedSettingsToolbar;
+
+        LinearLayout root = (LinearLayout) dialog.findViewById(android.R.id.list).getParent();
```

Comment:
I don't know. What's the problem? That `PreferenceActivity` does not have a toolbar because the layout is not provided the same way? What about using `PreferenceFragment` and a regular activity?